app/helpers/foreman_ansible/ansible_hostgroups_helper.rb

```diff
@@ -3,13 +3,9 @@
 module ForemanAnsible
   module AnsibleHostgroupsHelper
     def ansible_hostgroups_actions(hostgroup)
-      play_roles = if hostgroup.all_ansible_roles.empty?
-                     { action: (link_to _('Run all Ansible roles'), 'javascript:void(0);', disabled: true, title: 'No Roles assigned'), priority: 31 }
-                   else
-                     { action: display_link_if_authorized(_('Run all Ansible roles'), hash_for_play_roles_hostgroup_path(id: hostgroup), 'data-no-turbolink': true, title: _('Run all Ansible roles on hosts belonging to this host group')), priority: 31 }
-                   end
+      return [] if hostgroup.all_ansible_roles.empty? || !User.current.can?(:create_job_invocations)
 
-      [play_roles] if User.current.can?(:create_job_invocations)
+      [{ action: display_link_if_authorized(_('Run all Ansible roles'), hash_for_play_roles_hostgroup_path(id: hostgroup), 'data-no-turbolink': true, title: _('Run all Ansible roles on hosts belonging to this host group')), priority: 31 }]
```

Comment:
I like the code and it works as expected, however I'm not sure if removing the link is correct approach.
As a user, I find super confusing to have missing links just for some groups, maybe disabled link with hover help text explaining why it's disabled is better user experience.
I'm not UX guy, I would suggest to contact Maria Svirikova, she's the person to go for UX, she will know what to do.

Reply:
You are right, I'll contact Maria and ask her.

Reply:
Hi @MariSvirik, we want to make a UI change and your opinion could help us here.

On the Hostgroup page, we have the `Run all Ansible roles` option in the actions dropdown. 
However, not all Hostgroups have ansible roles assigned to them.
Currently if you hover over this option for a Hostgroup without any Ansible roles you'll see this tooltip:
![image](https://user-images.githubusercontent.com/38184193/185397717-1860bc38-cc6c-4597-acb0-1b4e6aa3a33f.png)

We are thinking about disabling or removing the `Run all Ansible roles` option from the dropdown for Hostgroups without any Ansible roles.

What do you think is the best approach here?

Reply:
Hi @nofaralfasi ,  @stejskalleos is right, the good practice is to disable the action and add a tooltip with an explanation. If the action could be enabled by users/their actions (which in this case is) and makes sense in the context of the page, this is the preferable approach.

Reply:
Great, thank you @MariSvirik! 
I will apply the new changes and check with you after to make sure it is working properly.

Reply:
Hi @MariSvirik, this is how the disabled action looks like with the new changes:
![Screenshot from 2022-08-23 17-45-28](https://user-images.githubusercontent.com/38184193/186188881-7bf299fa-a217-4d7c-9bac-9203b2734fa4.png)

Let me know if something is missing or needs to be changed.
Thanks!

Reply:
thank you for pinging me @nofaralfasi 
1. Remove the ban icon
2.  text should be either "No roles assigned" (sentence case) OR "To run all Ansible roles, firstly assign roles to the hostgroup." (where you explain how to enable the action)

- phrasing can be checked by the doc team (pinging them now)

Reply:
Great, I'll do the necessary changes.
Thank you for your help!

Reply:
Hi @MariSvirik, is there an update from the doc team regarding the phrasing?
Thanks!

Reply:
@MariSvirik 
Currently I only updated the text to sentence case "No roles assigned". If the text needs to be changed, please let me know. 
Thanks.